src/monitor.c

```diff
@@ -464,12 +465,7 @@ void monitor_init(void)
 	}
 }
 
-void monitor_run(iterator_t *it, pthread_mutex_t *lock)
-{
-	int_status_t *internal_status = xmalloc(sizeof(int_status_t));
-	export_status_t *export_status = xmalloc(sizeof(export_status_t));
-
-	while (!(zsend.complete && zrecv.complete)) {
+void export_then_update(int_status_t *internal_status, iterator_t *it, export_status_t *export_status, pthread_mutex_t *lock) {
```

Comment:
I have now encapsulated the main functionality within the while loop (which involves obtaining the current scanning status and writing it to a status file) into a separate function called `export_then_update`. This encapsulation simplifies the logic in `monitor_run`.

The structure of the while loop remains unchanged. The only addition is to call `export_then_update` once more after the while loop concludes. This ensures that the status file is updated based on the latest state.

If there are any concerns regarding the style or functionality of the code modifications I've made, please let me know. Your feedback is greatly appreciated! @droe

Reply:
This does address the race I pointed out.  No other concerns, I'm not a reviewer tho.